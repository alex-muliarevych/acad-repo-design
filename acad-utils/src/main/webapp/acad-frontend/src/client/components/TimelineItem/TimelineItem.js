import React from 'react';
import PropTypes from 'prop-types';

// Helpers
import BEM from 'helpers/BEM';
import { getFormattedHour } from 'helpers/DateHelpers';

// Styles
import './TimelineItem.less';

const bTimeItem = BEM.b('timeline-item');

export default class TimelineItem extends React.Component {

  render() {
    return (
      <div className={ bTimeItem() }>
        <span className={ bTimeItem('time-label') }>
          { getFormattedHour(this.props.hour) }
        </span>

        <div className={ bTimeItem('activities') }>
          { this.props.children }
        </div>
      </div>
    );
  }
}

TimelineItem.propTypes = {
  hour: PropTypes.number.isRequired,
  children: PropTypes.any
};

TimelineItem.defaultProps = {
  hour: 0,
};
