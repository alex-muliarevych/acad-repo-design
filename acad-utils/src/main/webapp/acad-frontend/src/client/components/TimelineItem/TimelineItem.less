@import (reference) '~styles/variables.less';

.timeline-item {
  position: relative;
  display: flex;
  margin-bottom: 10px;
  min-height: 50px;

  &__time-label {
    color: @dark-grey;
    margin-right: 10px;
  }

  &__activities {
    display: flex;
    flex-direction: column;
  }
}
