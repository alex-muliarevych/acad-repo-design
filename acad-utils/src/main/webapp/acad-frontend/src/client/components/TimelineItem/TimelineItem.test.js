import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import TimelineItem from './TimelineItem';

describe('TimelineItem', () => {
  let component;

  beforeEach(() => {
    const item = { hour: 1 };

    component = shallow(
      <TimelineItem
        hour={ item.hour }
        activities={ [] } />
    );
  });

  it(`should be truthy`, () => {
    expect(component).toBeTruthy();
  });

  it(`should render correctly`, () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
