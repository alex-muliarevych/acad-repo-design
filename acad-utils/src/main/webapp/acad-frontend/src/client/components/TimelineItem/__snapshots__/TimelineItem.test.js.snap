// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineItem should render correctly 1`] = `null`;
